sources/models/build.rs

```diff
@@ -4,62 +4,65 @@
+    println!("cargo:rerun-if-changed={}", MOD_LINK);
+
+    // This build.rs runs once as a build-dependency of storewolf, and again as a (regular)
+    // dependency of storewolf.  There's no reason to do this work twice.
+    if env::var("CARGO_CFG_TARGET_VENDOR").unwrap_or_else(|_| String::new()) == "bottlerocket" {
+        println!("cargo:warning=Already ran model build.rs for host, skipping for target");
+        process::exit(0);
     }
-    // Link to requested target
-    symlink(&target, &link)
+
+    generate_readme();
+    link_current_variant();
 }
 
 fn link_current_variant() {
     // The VARIANT variable is originally BUILDSYS_VARIANT, set in the top-level Makefile.toml,
     // and is passed through as VARIANT by the top-level Dockerfile.  It represents which OS
     // variant we're building, and therefore which API model to use.
-    let var = "VARIANT";
-    println!("cargo:rerun-if-env-changed={}", var);
-    let variant = env::var(var).unwrap_or_else(|_| {
-        eprintln!("For local builds, you must set the {} environment variable so we know which API model to build against.  Valid values are the directories in variants/, for example \"aws-k8s-1.17\".", var);
+    let variant = env::var(VARIANT_ENV).unwrap_or_else(|_| {
+        eprintln!("For local builds, you must set the {} environment variable so we know which API model to build against.  Valid values are the directories in variants/, for example \"aws-k8s-1.17\".", VARIANT_ENV);
         process::exit(1);
     });
 
     // Point to the source for the requested variant
-    let variant_link = "src/variant/current";
     let variant_target = format!("../{}", variant);
 
     // Make sure requested variant exists
     let variant_path = format!("src/{}", variant);
     if !Path::new(&variant_path).exists() {
-        eprintln!("The environment variable {} should refer to a directory under sources/models/src with an API model, but it's set to '{}' which doesn't exist", var, variant);
+        eprintln!("The environment variable {} should refer to a directory under sources/models/src with an API model, but it's set to '{}' which doesn't exist", VARIANT_ENV, variant);
         process::exit(1);
     }
 
     // Create the symlink for the following `cargo build` to use for its source code
-    symlink_force(&variant_target, variant_link).unwrap_or_else(|e| {
-        eprintln!("Failed to create symlink at '{}' pointing to '{}' - we need this to support different API models for different variants.  Error: {}", variant_link, variant_target, e);
+    symlink_safe(&variant_target, VARIANT_LINK).unwrap_or_else(|e| {
+        eprintln!("Failed to create symlink at '{}' pointing to '{}' - we need this to support different API models for different variants.  Error: {}", VARIANT_LINK, variant_target, e);
         process::exit(1);
     });
 
     // Also create the link for mod.rs so Rust can import source from the "current" link
     // created above.
```

Comment:
Does this comment need to move above `symlink_safe(&mod_target, MOD_LINK).unwrap_or_else(|e| {` ? Maybe its okay here, comment for block of code below.

Reply:
Yeah, it's describing the block below it that creates the mod.rs link; I wanted to explain what we're doing before we create the variable containing the target of the link.